### PR TITLE
Companionship feature

### PR DESCRIPTION
**Pull request by wiesbadener**

### CODE REVIEW

File: Simulator/simulator/sim_globals.py [hunk 1]
```diff
@@ -3,15 +3,15 @@
 Global simulator/garden constants. Affects both test run and RL settings
 """
 MAX_WATER_LEVEL = float(os.getenv('MAX_WATER_LEVEL', 0.5))  # % Volumetric Water Content = cubic meter water per grid point; maximal soil moisture capacity
-IRRIGATION_AMOUNT = float(os.getenv('IRRIGATION_AMOUNT', 0.001))  # cubic metre water -> 1 liter
+IRRIGATION_AMOUNT = float(os.getenv('IRRIGATION_AMOUNT', 0.002))  # cubic metre water -> 1 liter
```
Comment (w07wong) [inline Simulator/simulator/sim_globals.py:6]: Why the increase to 0.002?

File: Simulator/simulator/plant_presets.py [hunk 2]
```diff
@@ -64,58 +70,149 @@ def _compute_from_table_values(
...
-                                             germination_time=(11, 1),
-                                             seed_spacing=11, maturation_time=(60, 1), stopping_color=(202/255, 129/255, 1),
-                                             color_step=(10/255, -10/255, 0/255)),
-    "marigold": _compute_from_table_values(name="marigold", color=(117/255, 128/255, 81/255), germination_time=(7.5, 1),
-                                           seed_spacing=7, maturation_time=(50, 1), stopping_color=(177/255, 98/255, 1),
-                                           color_step=(10/255, -5/255, 0/255)),
-    # "calendula": _compute_from_table_values(name="calendula", color=(62/255, 129/255, 78/255), germination_time=(8.5, 1),
-    #                                         seed_spacing=12, maturation_time=(50, 1), stopping_color=(182/255, 129/255, 1)),
+    # https://www.gardeningknowhow.com/edible/herbs/borage/borage-herb.htm
+    "borage": _compute_from_table_values(name="borage", color=[(9 / 255, 77 / 255, 10 / 255),(0.9467, 0.6863, 0.2431)][SEG_COLORS], germination_time=(8, 5),
```
Comment (w07wong) [inline Simulator/simulator/plant_presets.py:109]: Just checking that you guys want std_dev 5 and 1 for all germination and maturation times respectively correct?

File: Simulator/simulator/plant_type.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import numpy as np
 from simulator.plant import Plant
-from simulator.plant_presets import PLANT_TYPES, generate_c1_and_growth_time
+from simulator.plant_presets import PLANT_TYPES, PLANTS_RELATION, generate_c1_and_growth_time
 from simulator.sim_globals import NUM_PLANTS, NUM_PLANT_TYPES_USED
+import pickle
+import math
```
Comment (w07wong) [inline Simulator/simulator/plant_type.py:6]: Unused import?

File: Simulator/simulator/garden.py [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 from heapq import nlargest
 from simulator.logger import Logger, Event
-#from simulator.visualization import setup_animation, setup_saving
-from simulator.sim_globals import MAX_WATER_LEVEL, IRRIGATION_AMOUNT, PERMANENT_WILTING_POINT, PRUNE_DELAY, PRUNE_THRESHOLD, NUM_IRR_ACTIONS, PRUNE_RATE
+# from simulator.visualization import setup_animation, setup_saving
+from simulator.sim_globals import MAX_WATER_LEVEL, IRRIGATION_AMOUNT, PERMANENT_WILTING_POINT, PRUNE_DELAY, \
+    PRUNE_THRESHOLD, NUM_IRR_ACTIONS, PRUNE_RATE
 import pickle
 import multiprocessing as mp
+import os
```
Comment (w07wong) [inline Simulator/simulator/garden.py:9]: Is this used?

File: Simulator/simulator/sim_globals.py (hunk 1 above)
Comment (wiesbadener) [reply Simulator/simulator/sim_globals.py:6]: Experiments were conducted with 0.002

File: Simulator/simulator/plant_presets.py (hunk 2 above)
Comment (wiesbadener) [reply Simulator/simulator/plant_presets.py:109]: @mpresten is this correct?

Comment (mpresten) [reply Simulator/simulator/plant_presets.py:109]: @wiesbadener I'm adjusting it to be plant specific for germ_time. We could just have it be 5 for all maturation_times though

File: Simulator/simulator/visualizer.py [hunk 5]
```diff
@@ -137,10 +137,10 @@ class Pillow_Visualizer(Visualizer):
     def __init__(self, env):
         super().__init__(env)
 
-    def get_canopy_image(self, bounds, dir_path, eval, scale=8, identifier="test"):
+    def get_canopy_image(self, bounds, dir_path, eval, scale=8, identifier="test", day=0):
         x_low, y_low, x_high, y_high = bounds
         row_scale, col_scale = (self.env.rows // (x_high - x_low)) * scale, (self.env.cols // (y_high - y_low)) * scale
-        image = Image.new('RGB', (self.env.cols * col_scale, self.env.rows * row_scale), (255, 255, 255))
+        image = Image.new('RGBA', (self.env.cols * col_scale, self.env.rows * row_scale), (255, 255, 255, 0))
```
Comment (wiesbadener) [inline Simulator/simulator/visualizer.py:143]: @w07wong are you ok with this?

File: Simulator/simulator/sim_globals.py (hunk 1 above)
Comment (w07wong) [reply Simulator/simulator/sim_globals.py:6]: Real world experiments?

File: Simulator/simulator/visualizer.py (hunk 5 above)
Comment (w07wong) [reply Simulator/simulator/visualizer.py:143]: Yeah we don't use visualizer for data_collection if that's what you mean.  If the code works its fine.

File: Simulator/simulator/garden.py (hunk 4 above)
Comment (mpresten) [reply Simulator/simulator/garden.py]: I don't believe so, I used this for growth rate analysis

**Review by w07wong — APPROVED** [review]

**Review by w07wong — APPROVED** [review]

**Pull request merged**